hypertrace-trace-enricher/hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/enrichers/HttpAttributeEnricher.java

```diff
@@ -78,12 +78,22 @@ public void enrichEvent(StructuredTrace trace, Event event) {
         //split only on first occurrence of delimiter. eg: cat=1dog=2 should be split to cat -> 1dog=2
         .map(kv -> kv.split(QUERY_PARAM_KEY_VALUE_DELIMITER, 2))
         .map(kv -> Pair.of(
-            String.format(PARAM_ATTR_FORMAT, HTTP_REQUEST_QUERY_PARAM_ATTR, decode(kv[0])),
+            String.format(PARAM_ATTR_FORMAT, HTTP_REQUEST_QUERY_PARAM_ATTR, decodeParamKey(kv[0])),
             kv.length == 2 ? decode(kv[1]) : ""
         ))
         .collect(groupingBy(Pair::getKey, mapping(Pair::getValue, toList())));
   }
 
+  private String decodeParamKey(String input) {
```

Comment:
I feel like this could perfectly be static and although that would drag `decode` to be static too it is worth to try it.

Reply:
makes sense, done.